chemprop/v2/utils/v1_to_v2.py

```diff
@@ -0,0 +1,140 @@
+    d_h, d_e = model_v1["state_dict"]["encoder.encoder.0.W_i.weight"].shape
+    d_v = model_v1["state_dict"]["encoder.encoder.0.W_o.weight"].shape[1] - d_h
+    hyper_parameters_v2["message_passing"] = AttributeDict(
+        {
+            "activation": args_v1.activation,
+            "bias": args_v1.bias,
+            "cls": BondMessageBlock if not model_v1['args'].atom_messages else AtomMessageBlock,
+            "d_e": d_e,  # the feature dimension of the edges
+            "d_h": model_v1['args'].hidden_size,  # dimension of the hidden layer
+            "d_v": d_v,  # the feature dimension of the vertices
+            "d_vd": None,  # ``d_vd`` is the number of additional features that will be concatenated to atom-level features *after* message passing
+            "depth": args_v1.depth,
+            "dropout": args_v1.dropout,
+            "undirected": args_v1.undirected,
+        }
+    )
+
+    # convert the aggregation block
+    hyper_parameters_v2["agg"] = {
+        "dim": 0, # in v1, the aggregation is always done on the atom features
+        "cls": AggregationRegistry[args_v1.aggregation],
+    }
+    if args_v1.aggregation == "norm":
+        hyper_parameters_v2["agg"]["norm"] = args_v1.aggregation_norm
+
+    # convert the readout block
+    hyper_parameters_v2["readout"] = AttributeDict({
+        "activation": args_v1.activation,
+        "cls": ReadoutRegistry[args_v1.dataset_type],
+        "criterion": LossFunctionRegistry[args_v1.loss_function],
+        "dropout": args_v1.dropout,
+        "hidden_dim": args_v1.ffn_hidden_size,
+        "input_dim": args_v1.hidden_size,
+        "loc": model_v1["data_scaler"]["means"][0],
+        "n_layers": args_v1.ffn_num_layers-1,
+        "n_tasks": args_v1.num_tasks,
+        "scale": model_v1["data_scaler"]["stds"][0],
+    })
+
+    return hyper_parameters_v2
+
+
+def convert_model_v1_to_v2(model_v1: dict) -> dict:
+    """Converts a v1 model dictionary from a loaded .pt file to a v2 model dictionary
+    """
+    
+    model_v2 = {}
+    
+    model_v2["epoch"] = None  # Not defined in v1 model dictionary
+    model_v2["global_step"] = None  # Not defined in v1 model dictionary
+    model_v2["pytorch-lightning_version"] = __version__ 
+    model_v2["state_dict"] = convert_state_dict_v1_to_v2(model_v1)
+    model_v2["loops"] = None  # Not defined in v1 model dictionary
+    model_v2["callbacks"] = None  # Not defined in v1 model dictionary
+    model_v2["optimizer_states"] = None  # Not defined in v1 model dictionary
+    model_v2["lr_schedulers"] = None  # Not defined in v1 model dictionary
+    model_v2["hparams_name"] = "kwargs"
+    model_v2["hyper_parameters"] = convert_hyper_parameters_v1_to_v2(model_v1)
+
+    return model_v2
+
+def convert_model_v1_to_v2_file(model_v1_file: str, model_v2_file: str) -> None:
+    """Converts a v1 model .pt file to a v2 model .ckpt file
+    """
+    model_v1 = torch.load(model_v1_file)
+    model_v2 = convert_model_v1_to_v2(model_v1)
```

Comment:
can we make the variable names more clear? `model_v*` makes it sound like these variables are actual models/DMPNNs, but these are just dictionaries containing model (hyper)parameters?